op-batcher/batcher/driver.go

```diff
@@ -543,14 +546,26 @@ func (l *BatchSubmitter) throttlingLoop(throttlingLoopDone chan struct{}) {
 				maxBlockSize = l.Config.ThrottleBlockSize
 			}
 		}
-		var success bool
+		var (
+			success bool
+			rpcErr  rpc.Error
+		)
 		if err := cl.Client().CallContext(
-			ctx, &success, SetMaxDASizeMethod, hexutil.Uint64(maxTxSize), hexutil.Uint64(maxBlockSize)); err != nil {
-			l.Log.Error("SetMaxDASize rpc failed", "err", err)
+			ctx, &success, SetMaxDASizeMethod, hexutil.Uint64(maxTxSize), hexutil.Uint64(maxBlockSize),
+		); errors.As(err, &rpcErr) && eth.ErrorCode(rpcErr.ErrorCode()).IsGenericRPCError() {
+			l.Log.Error("SetMaxDASize rpc unavailable or broken, shutting down. Either enable it or disable throttling.", "err", err)
+			// We'd probably hit this error right after startup, so a short shutdown duration should suffice.
+			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+			defer cancel()
+			// Always returns nil. An error is only returned to expose this function as an RPC.
+			_ = l.StopBatchSubmitting(ctx)
+			return
```

Comment:
Will this cause the `throttlingLoop` to return?

Reply:
No, this just returns the current `updateParams()` call. It's the `l.StopBatchSubmitting` call that shuts down the whole batcher, and in particular also the throttling loop.